app/cells/decidim/alternative_landing/content_blocks/base_cell.rb

```diff
@@ -18,14 +18,22 @@ def participatory_spaces
           ].flatten.compact
         end
 
-        def available_components(manifest_name)
-          Decidim::Component.published.where(participatory_space: participatory_spaces, manifest_name: manifest_name).map do |component|
+        def available_components
+          components.where(manifest_name: manifest_name).map do |component|
             ["#{translated_attribute(component.name)} (#{translated_attribute(component.participatory_space.title)})", component.id]
           end.unshift [t(".all"), nil]
         end
 
         def component
-          @component ||= Decidim::Component.find_by(id: form.object.settings.try(:component_id))
+          components.find_by(id: (defined?(form) ? form.object : model).settings.try(:component_id))
+        end
+
+        def components
+          @components ||= Decidim::Component.where(participatory_space: participatory_spaces)
```

Comment:
```suggestion
          Decidim::Component.where(participatory_space: participatory_spaces)
```
Not sure memoize does anything here

Reply:
Does it hurt, though? 🤔

Reply:
sure not